hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcClient.java

```diff
@@ -462,7 +481,8 @@ private RetryDecision shouldRetry(final IOException ioe, final int retryCount,
   public Object invokeMethod(
       final UserGroupInformation ugi,
       final List<? extends FederationNamenodeContext> namenodes,
-      final Class<?> protocol, final Method method, final Object... params)
+      final Class<?> protocol, final Method method, boolean skipObserver,
```

Comment:
It's a nit, but I'd move skipObserver up next to the namenodes. At the higher level we use useObserver and negate when calling here. We should probably be consistent.